agent/lib/kontena/workers/log_worker.rb

```diff
@@ -56,22 +58,34 @@ def watch_queue
       end
     end
 
-    # Process items from @queue until nil
+    # Process items from @queue
     def process_queue
-      defer {
-        while data = @queue.pop
-          sleep 1 until @processing
-
-          rpc_client.async.notification('/containers/log', [data])
-          if @queue.size > 100
-            sleep 0.001
-          else
-            sleep 0.05
-          end
+      while buffer = @queue.shift(BATCH_SIZE)
```

Comment:
This while loop condition doesn't make much sense anymore, I don't think it's possible for `@queue.shift(n)` to return anything falsey

Reply:
Changed to `loop`.